apps/aecore/lib/aecore/miner/worker.ex

```diff
@@ -240,31 +226,26 @@ defmodule Aecore.Miner.Worker do
                       top_block.header.height + 1 +
                       Application.get_env(:aecore, :tx_data)[:lock_time_coinbase]) |
                    valid_txs_by_fee]
-      root_hash = BlockValidation.calculate_root_hash(valid_txs)
-
-      new_chain_state = ChainState.calculate_and_validate_chain_state!(valid_txs, chain_state, top_block.header.height + 1)
-      chain_state_hash = ChainState.calculate_chain_state_hash(new_chain_state)
-
-      top_block_hash = BlockValidation.block_header_hash(top_block.header)
-
-      unmined_header =
-        Header.create(
-          top_block.header.height + 1,
-          top_block_hash,
-          root_hash,
-          chain_state_hash,
-          difficulty,
-          0,
-          #start from nonce 0, will be incremented in mining
-          Block.current_block_version()
-        )
-      %Block{header: unmined_header, txs: valid_txs}
+
+      new_block = create_block(top_block, chain_state, difficulty, [])
+      new_block_size_bits = new_block |> :erlang.term_to_binary() |> Bits.extract() |> Enum.count()
+      new_block_size_bytes = new_block_size_bits / 8
+
+      valid_txs_by_block_size = filter_transactions_by_block_size(valid_txs, new_block_size_bytes)
+
+      total_fees = calculate_total_fees(valid_txs_by_block_size)
+      valid_txs =
```

Comment:
Why this replace?
`Filter_transactions_by_block_size` should preserve (and it looks like it preserves) order and remove transactions from the end, so the coinbase tx would never get removed as that's the first one.

Reply:
The replace is because we recalculate `total_fees`, based on the filtered transactions by block size.